tsl/src/continuous_aggs/create.c

```diff
@@ -786,6 +867,37 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errmsg("only immutable expressions allowed in time bucket function"),
 						 errhint("Use an immutable expression as first argument"
 								 " to the time bucket function.")));
+
+			if ((tbinfo->bucket_width == BUCKET_WIDTH_VARIABLE) && (tbinfo->interval->month != 0))
+			{
+				/* Monthly buckets case */
+				if (!TIMESTAMP_NOT_FINITE(tbinfo->origin))
+				{
+					/*
+					 * Origin was specified - make sure it's the first day of the month.
+					 * If a timezone was specified the check should be done in this timezone.
+					 */
+					Timestamp origin = tbinfo->origin;
+					if (tbinfo->timezone != NULL)
+					{
+						/* The code is equal to 'timestamptz AT TIME ZONE tzname'. */
+						origin = DatumGetTimestamp(
+							DirectFunctionCall2(timestamptz_zone,
+												CStringGetTextDatum(tbinfo->timezone),
+												TimestampTzGetDatum((TimestampTz) origin)));
+					}
+
+					const char *day =
+						TextDatumGetCString(DirectFunctionCall2(timestamp_to_char,
+																TimestampGetDatum(origin),
+																CStringGetTextDatum("DD")));
+					if (strcmp(day, "01") != 0)
+						ereport(ERROR,
+								(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+								 errmsg("for monthly buckets origin must be the first day of the "
```

Comment:
Why is that?

Reply:
For instance, if you specify origin = '2022-01-29' adding 1 month will result in '2022-02-29', but there is no such date. It's unclear how to handle such cases, thus currently we explicitly forbid doing this. See also the tests for time_bucket_ng().

Reply:
Ok I guess, but I wonder if users expect to do every 15th of the month with origin for example.

Reply:
Sure. We just didn't want to implement this in MVP thus currently it's not supported by time_bucket_ng() 🤷‍♂️ If there will be a demand from the users it will be not difficult to deliver.